src/map.rs

```diff
@@ -168,6 +160,39 @@ impl<'a, T> PutResult<'a, T> {
     }
 }
 
+/// The error type for the [`HashMap::try_insert`] method.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct TryInsertError<'a, V> {
+    /// A reference to the old value mapped to the key.
+    pub old: &'a V,
+    /// A reference to the new value, the one passed as an argument to
```

Comment:
I would avoid referring to this as "new" in this context, since it's not new as we did not do the replace. Maybe
> The value that [`HashMap::try_insert`] failed to insert.